tests/test_config.py

```diff
@@ -0,0 +1,140 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+import argparse
+
+import importlib
+import inspect
+
+from pathlib import Path
+from unittest.mock import patch
+
+import pytest
+
+import torch
+import torchtune
+
+from omegaconf import OmegaConf
+
+from tests.test_utils import gpu_test, TOKENIZER_PATHS
+from torchtune import config, training
+from torchtune._recipe_registry import get_all_recipes
+from torchtune.config._utils import _merge_yaml_and_cli_args
+
+ROOT = Path(torchtune.__file__).parent.parent
+
+RECIPE_ROOT = ROOT / "recipes"
+CONFIG_ROOT = ROOT / "recipes" / "configs"
+
+
+def _get_all_configs(recipe_types):
+    recipes = get_all_recipes()
+    return [
+        (recipe.file_path, config.file_path)
+        for recipe in recipes
+        if recipe.name in recipe_types
+        for config in recipe.configs
+    ]
+
+
+def load_module_from_path(module_name, path):
+    spec = importlib.util.spec_from_file_location(module_name, path)
+    module = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(module)
+    return module
+
+
+class TestRecipeConfigs:
+    def validate_tokenizer(self, tokenizer_cfg):
+        with pytest.raises(OSError, match="No such file or directory"):
```

Comment:
Actually maybe this test is stupid because it doesn't validate anything after a given OSError being raised. I should instead mock the load function and just make sure it instantiates OK.